subproviders/filters.js

```diff
@@ -345,7 +345,8 @@ function LogFilter(opts) {
   self.type = 'log'
   self.fromBlock = (opts.fromBlock !== undefined) ? opts.fromBlock : 'latest'
   self.toBlock = (opts.toBlock !== undefined) ? opts.toBlock : 'latest'
-  self.address = opts.address ? normalizeHex(opts.address) : opts.address
+  expectedAddress = opts.address && (Array.isArray(opts.address) ? opts.address : [opts.address]);
+  self.address = expectedAddress && expectedAddress.map(function(a) { return normalizeHex(a) });
```

Comment:
this can be shortened as well

`self.address = expectedAddress && expectedAddress.map(normalizeHex)`